Core/include/Acts/EventData/Measurement.hpp

```diff
@@ -453,6 +453,7 @@ class FixedSizeMeasurement {
 
   /// Construct from source link, subset indices, and measured data.
   ///
+  /// @tparam index_container_t Forward-iterable Container for indices
```

Comment:
Can you elaborate why this is done now?

Reply:
I saw that it is inconvenient to require the input to be a `std::array`. E.g. the input indices might be stored in a `std::vector` as they are provided only at runtime. To avoid having to always convert manually to a `std::array`, I take an arbitrary container for better usability.